sov-rollup-starter/src/rollup.rs

```diff
@@ -1,110 +1,99 @@
+    type DaSpec = MockDaSpec;
+    type DaConfig = MockDaConfig;
+
+    fn create_genesis_config(
+        &self,
+        genesis_paths: &Self::GenesisPaths,
+        rollup_config: &RollupConfig<Self::DaConfig>,
+    ) -> <Self::NativeRuntime as RuntimeTrait<Self::NativeContext, Self::DaSpec>>::GenesisConfig
+    {
+        get_genesis_config(rollup_config.da.sender_address, genesis_paths)
+    }
+
+    async fn create_da_service(
+        &self,
+        rollup_config: &RollupConfig<Self::DaConfig>,
+    ) -> Self::DaService {
+        MockDaService::new(rollup_config.da.sender_address)
+    }
+
+    fn create_vm(&self) -> Self::Vm {
+        Risc0Host::new(risc0_starter::MOCK_DA_ELF)
+    }
 
-impl<Vm: ZkvmHost, Da: DaService<Error = anyhow::Error> + Clone> Rollup<Vm, Da> {
-    /// Creates a new rollup instance
-    #[allow(clippy::type_complexity)]
-    pub fn new<DaConfig: DeserializeOwned>(
-        da_service: Da,
-        genesis_config: GenesisConfig<DefaultContext, Da::Spec>,
-        config: RollupConfig<DaConfig>,
-        prover: Option<Prover<ZkStf<Da::Spec, Vm::Guest>, Da, Vm>>,
-    ) -> Result<Self, anyhow::Error> {
-        let ledger_db = LedgerDB::with_path(&config.storage.path)?;
-        let app = StfWithBuilder::new(config.storage.clone());
-        Ok(Self {
-            app,
-            da_service,
-            ledger_db,
-            runner_config: config.runner,
-            genesis_config,
-            prover,
-        })
+    fn create_zk_storage(
+        &self,
+        _rollup_config: &RollupConfig<Self::DaConfig>,
+    ) -> <Self::ZkContext as Spec>::Storage {
+        ZkStorage::new()
     }
 
-    /// Runs the rollup.
-    pub async fn run(self) -> Result<(), anyhow::Error> {
-        self.run_and_report_rpc_port(None).await
+    fn create_verifier(&self) -> <Self::DaService as DaService>::Verifier {
+        Default::default()
     }
 
-    /// Runs the rollup. Reports rpc port to the caller using the provided channel.
-    pub async fn run_and_report_rpc_port(
-        mut self,
-        channel: Option<oneshot::Sender<std::net::SocketAddr>>,
-    ) -> Result<(), anyhow::Error> {
-        let storage = self.app.get_storage();
-        let last_slot_opt = self.ledger_db.get_head_slot()?;
-        let prev_root = last_slot_opt
-            .map(|(number, _)| storage.get_root_hash(number.0))
-            .transpose()?;
-
-        let rpc_module = self.rpc_module(storage)?;
-
-        let mut runner = StateTransitionRunner::new(
-            self.runner_config,
-            self.da_service,
-            self.ledger_db,
-            self.app.stf,
-            prev_root,
-            self.genesis_config,
-            self.prover,
-        )?;
-
-        runner.start_rpc_server(rpc_module, channel).await;
-        runner.run_in_process().await?;
-
-        Ok(())
+    fn create_native_storage(
+        &self,
+        rollup_config: &RollupConfig<Self::DaConfig>,
+    ) -> <Self::NativeContext as Spec>::Storage {
```

Comment:
This function handles I/O and should probably be fallible

Reply:
done